Mathlib/Order/JordanHolder.lean

```diff
@@ -303,298 +210,121 @@ theorem ext {s₁ s₂ : CompositionSeries X} (h : ∀ x, x ∈ s₁ ↔ x ∈ s
-#align composition_series.bot_mem CompositionSeries.bot_mem
+#align composition_series.bot RelSeries.head
+#align composition_series.bot_mem RelSeries.head_mem
 
 @[simp]
-theorem bot_le {s : CompositionSeries X} (i : Fin (s.length + 1)) : s.bot ≤ s i :=
+theorem head_le {s : CompositionSeries X} (i : Fin (s.length + 1)) : s.head ≤ s i :=
   s.strictMono.monotone (Fin.zero_le _)
-#align composition_series.bot_le CompositionSeries.bot_le
+#align composition_series.bot_le CompositionSeries.head_le
 
-theorem bot_le_of_mem {s : CompositionSeries X} {x : X} (hx : x ∈ s) : s.bot ≤ x :=
+theorem head_le_of_mem {s : CompositionSeries X} {x : X} (hx : x ∈ s) : s.head ≤ x :=
   let ⟨_i, hi⟩ := Set.mem_range.2 hx
-  hi ▸ bot_le _
-#align composition_series.bot_le_of_mem CompositionSeries.bot_le_of_mem
-
-theorem length_pos_of_mem_ne {s : CompositionSeries X} {x y : X} (hx : x ∈ s) (hy : y ∈ s)
-    (hxy : x ≠ y) : 0 < s.length :=
-  let ⟨i, hi⟩ := hx
-  let ⟨j, hj⟩ := hy
-  have hij : i ≠ j := mt s.inj.2 fun h => hxy (hi ▸ hj ▸ h)
-  hij.lt_or_lt.elim
-    (fun hij => lt_of_le_of_lt (zero_le (i : ℕ)) (lt_of_lt_of_le hij (Nat.le_of_lt_succ j.2)))
-      fun hji => lt_of_le_of_lt (zero_le (j : ℕ)) (lt_of_lt_of_le hji (Nat.le_of_lt_succ i.2))
-#align composition_series.length_pos_of_mem_ne CompositionSeries.length_pos_of_mem_ne
-
-theorem forall_mem_eq_of_length_eq_zero {s : CompositionSeries X} (hs : s.length = 0) {x y}
-    (hx : x ∈ s) (hy : y ∈ s) : x = y :=
-  by_contradiction fun hxy => pos_iff_ne_zero.1 (length_pos_of_mem_ne hx hy hxy) hs
-#align composition_series.forall_mem_eq_of_length_eq_zero CompositionSeries.forall_mem_eq_of_length_eq_zero
-
-/-- Remove the largest element from a `CompositionSeries`. If the series `s`
-has length zero, then `s.eraseTop = s` -/
-@[simps]
-def eraseTop (s : CompositionSeries X) : CompositionSeries X where
-  length := s.length - 1
-  series i := s ⟨i, lt_of_lt_of_le i.2 (Nat.succ_le_succ tsub_le_self)⟩
-  step' i := by
-    have := s.step ⟨i, lt_of_lt_of_le i.2 tsub_le_self⟩
-    cases i
-    exact this
-#align composition_series.erase_top CompositionSeries.eraseTop
-
-theorem top_eraseTop (s : CompositionSeries X) :
-    s.eraseTop.top = s ⟨s.length - 1, lt_of_le_of_lt tsub_le_self (Nat.lt_succ_self _)⟩ :=
-  show s _ = s _ from
-    congr_arg s
-      (by
-        ext
-        simp only [eraseTop_length, Fin.val_last, Fin.coe_castSucc, Fin.coe_ofNat_eq_mod,
-          Fin.val_mk])
-#align composition_series.top_erase_top CompositionSeries.top_eraseTop
+  hi ▸ head_le _
+#align composition_series.bot_le_of_mem CompositionSeries.head_le_of_mem
 
-theorem eraseTop_top_le (s : CompositionSeries X) : s.eraseTop.top ≤ s.top := by
-  simp [eraseTop, top, s.strictMono.le_iff_le, Fin.le_iff_val_le_val, tsub_le_self]
-#align composition_series.erase_top_top_le CompositionSeries.eraseTop_top_le
+#noalign composition_series.length_pos_of_mem_ne
+#noalign composition_series.forall_mem_eq_of_length_eq_zero
```

Comment:
So again you're removing two lemmas completely. Should there be RelSeries versions of these?

Reply:
there is a similar version in `RelSeries`, similar but not the same. I aligned them anyway and added a note

Reply:
I'm confused -- they're still #noalign

Reply:
My bad, I think I messed up this when dealing with merge conflict. I have aligned them again.